DM-39918: Initial version

rra opened the pull request:
Initial version of tech note on Phalanx secrets management design.

rra committed:
Metadata and current situation

Flesh out the metadata and start with a description of the current
secrets management scheme.

rra committed:
Add some requirements

Flesh out the current state and add a diagram. Add requirments for
the secret management system, and start describing the application
secret model.

rra committed:
Flesh out the proposed design

rra committed:
Finish proposed design

rra committed:
Spell check

rra committed:
Fix link to Argo CD

rra committed:
Update diagram labels

jonathansick commented:
For 1Password item naming conventions, would it be possible to colocate all of an app's secrets in a single secret item, and then use the named fields in that 1Password item to correspond to the individual secret fields? I think the "Server" item type can be totally customized, so that all fields can be customized.

frossie commented:
Looks great. no notes

rra committed:
Update 1Password, secret spec

Instead of using separate 1Password entries per secret, use one
entry per application and make use of the complete customization
of the Server type so that all the key and value pairs can be the
static secrets for that application.

Document special handling for pull-secret and improve the
documentation for the Vault read secret.

Improve the secrets.yaml specification to allow the generate and
copy stanzas to also be conditional.

rra commented:
> For 1Password item naming conventions, would it be possible to colocate all of an app's secrets in a single secret item, and then use the named fields in that 1Password item to correspond to the individual secret fields? I think the "Server" item type can be totally customized, so that all fields can be customized.

This is a great idea, thanks. Now included.
